crowbar_framework/app/controllers/ceph_pre_upgrade_controller.rb

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright 2017, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+class CephPreUpgradeController < ApplicationController
+  def index
+    ceph_nodes = get_ceph_nodes
+    @prepared = nodes_prepared(ceph_nodes)
+    @no_ceph_nodes = ceph_nodes.empty?
+
+    respond_to do |format|
+      format.json do
+        render json: { prepared: @prepared, no_ceph_nodes: @no_ceph_nodes }
+      end
+      format.html
+    end
+  end
+
+  def prepare
+    status = :ok
+    error_msg = ""
+    prepared = nodes_prepared(get_ceph_nodes)
+
+    begin
+      service_object = CrowbarService.new(Rails.logger)
+      if prepared
```

Comment:
You can add hidden field in the form e.g.: `%input{type: "hidden", name: "prepare_action", value: @prepared ? 'revert' : 'prepare'}`. This will be available as `params[:prepare_action]` inside the controller.